src/backend/plugins/config-kde/config-kde.c

```diff
@@ -0,0 +1,296 @@
+        self->http_proxy = g_strdup (value->str);
+      } else if (strcmp (kv[0], "ftpProxy") == 0) {
+        self->ftp_proxy = g_strdup (value->str);
+      } else if (strcmp (kv[0], "socksProxy") == 0) {
+        self->socks_proxy = g_strdup (value->str);
+      } else if (strcmp (kv[0], "NoProxyFor") == 0) {
+        g_autofree char *no_proxy_for = g_strdup (value->str);
+        self->no_proxy = g_strsplit (no_proxy_for, ",", -1);
+      } else if (strcmp (kv[0], "Proxy Config Script") == 0) {
+        self->pac_script = g_strdup (value->str);
+      } else if (strcmp (kv[0], "ProxyType") == 0) {
+        self->proxy_type = atoi (value->str);
+      }
+    }
+  } while (line);
+
+  self->available = TRUE;
+}
+
+
+static void
+px_config_kde_init (PxConfigKde *self)
+{
+  px_config_kde_set_config_file (self, NULL);
+}
+
+static void
+px_config_kde_dispose (GObject *object)
+{
+  PxConfigKde *self = PX_CONFIG_KDE (object);
+
+  g_clear_pointer (&self->config_file, g_free);
+  g_clear_object (&self->monitor);
+  g_clear_pointer (&self->no_proxy, g_strfreev);
+  g_clear_pointer (&self->http_proxy, g_free);
+  g_clear_pointer (&self->https_proxy, g_free);
+  g_clear_pointer (&self->ftp_proxy, g_free);
+  g_clear_pointer (&self->socks_proxy, g_free);
+  g_clear_pointer (&self->pac_script, g_free);
+
+  G_OBJECT_CLASS (px_config_kde_parent_class)->dispose (object);
+}
+
+static void
+px_config_kde_set_property (GObject      *object,
+                            guint         prop_id,
+                            const GValue *value,
+                            GParamSpec   *pspec)
+{
+  PxConfigKde *config = PX_CONFIG_KDE (object);
+
+  switch (prop_id) {
+    case PROP_CONFIG_OPTION:
+      px_config_kde_set_config_file (config, g_value_dup_string (value));
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+px_config_kde_get_property (GObject    *object,
+                            guint       prop_id,
+                            GValue     *value,
+                            GParamSpec *pspec)
+{
+  PxConfigKde *config = PX_CONFIG_KDE (object);
+
+  switch (prop_id) {
+    case PROP_CONFIG_OPTION:
+      g_value_set_string (value, config->config_file);
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+px_config_kde_class_init (PxConfigKdeClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->dispose = px_config_kde_dispose;
+  object_class->set_property = px_config_kde_set_property;
+  object_class->get_property = px_config_kde_get_property;
+
+  g_object_class_override_property (object_class, PROP_CONFIG_OPTION, "config-option");
+}
+
+static gboolean
+px_config_kde_is_available (PxConfig *config)
+{
+  PxConfigKde *self = PX_CONFIG_KDE (config);
+
+  return self->available && g_getenv ("KDE_FULL_SESSION") != NULL;
```

Comment:
Please check `XDG_CURRENT_DESKTOP` here for symmetry, at first this could be alongside `KDE_FULL_SESSION`

Reply:
Surely we don't need to check `KDE_FULL_SESSION` anymore? That sounds like cruft that predates `XDG_CURRENT_DESKTOP`?

Reply:
Reworked KDE availability check.